File: snippets/java/src/test/java/com/algolia/Analytics.java
package com.algolia.methods.snippets;

// >IMPORT
import com.algolia.api.AnalyticsClient;
import com.algolia.model.analytics.*;

// IMPORT<

class SnippetAnalyticsClient {

  // Snippet for the customDelete method.
  //
  // allow del method for a custom path with minimal parameters
  void snippetForCustomDelete() {
    // >SEPARATOR customDelete default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.customDelete("test/minimal");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the customGet method.
  //
  // allow get method for a custom path with minimal parameters
  void snippetForCustomGet() {
    // >SEPARATOR customGet default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.customGet("test/minimal");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the customPost method.
  //
  // allow post method for a custom path with minimal parameters
  void snippetForCustomPost() {
    // >SEPARATOR customPost default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.customPost("test/minimal");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the customPut method.
  //
  // allow put method for a custom path with minimal parameters
  void snippetForCustomPut() {
    // >SEPARATOR customPut default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.customPut("test/minimal");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getAddToCartRate method.
  //
  // get getAddToCartRate with minimal parameters
  void snippetForGetAddToCartRate() {
    // >SEPARATOR getAddToCartRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getAddToCartRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getAverageClickPosition method.
  //
  // get getAverageClickPosition with minimal parameters
  void snippetForGetAverageClickPosition() {
    // >SEPARATOR getAverageClickPosition default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getAverageClickPosition("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getClickPositions method.
  //
  // get getClickPositions with minimal parameters
  void snippetForGetClickPositions() {
    // >SEPARATOR getClickPositions default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getClickPositions("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getClickThroughRate method.
  //
  // get getClickThroughRate with minimal parameters
  void snippetForGetClickThroughRate() {
    // >SEPARATOR getClickThroughRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getClickThroughRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getConversionRate method.
  //
  // get getConversationRate with minimal parameters
  void snippetForGetConversionRate() {
    // >SEPARATOR getConversionRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getConversionRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getNoClickRate method.
  //
  // get getNoClickRate with minimal parameters
  void snippetForGetNoClickRate() {
    // >SEPARATOR getNoClickRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getNoClickRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getNoResultsRate method.
  //
  // get getNoResultsRate with minimal parameters
  void snippetForGetNoResultsRate() {
    // >SEPARATOR getNoResultsRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getNoResultsRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getPurchaseRate method.
  //
  // get getPurchaseRate with minimal parameters
  void snippetForGetPurchaseRate() {
    // >SEPARATOR getPurchaseRate default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getPurchaseRate("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getRevenue method.
  //
  // get getRevenue with minimal parameters
  void snippetForGetRevenue() {
    // >SEPARATOR getRevenue default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getRevenue("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getSearchesCount method.
  //
  // get getSearchesCount with minimal parameters
  void snippetForGetSearchesCount() {
    // >SEPARATOR getSearchesCount default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getSearchesCount("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getSearchesNoClicks method.
  //
  // get getSearchesNoClicks with minimal parameters
  void snippetForGetSearchesNoClicks() {
    // >SEPARATOR getSearchesNoClicks default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getSearchesNoClicks("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getSearchesNoResults method.
  //
  // get getSearchesNoResults with minimal parameters
  void snippetForGetSearchesNoResults() {
    // >SEPARATOR getSearchesNoResults default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getSearchesNoResults("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getStatus method.
  //
  // get getStatus with minimal parameters
  void snippetForGetStatus() {
    // >SEPARATOR getStatus default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getStatus("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopCountries method.
  //
  // get getTopCountries with minimal parameters
  void snippetForGetTopCountries() {
    // >SEPARATOR getTopCountries default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopCountries("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopFilterAttributes method.
  //
  // get getTopFilterAttributes with minimal parameters
  void snippetForGetTopFilterAttributes() {
    // >SEPARATOR getTopFilterAttributes default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopFilterAttributes("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopFilterForAttribute method.
  //
  // get getTopFilterForAttribute with minimal parameters
  void snippetForGetTopFilterForAttribute() {
    // >SEPARATOR getTopFilterForAttribute default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopFilterForAttribute("myAttribute", "index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopFiltersNoResults method.
  //
  // get getTopFiltersNoResults with minimal parameters
  void snippetForGetTopFiltersNoResults() {
    // >SEPARATOR getTopFiltersNoResults default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopFiltersNoResults("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopHits method.
  //
  // get getTopHits with minimal parameters
  void snippetForGetTopHits() {
    // >SEPARATOR getTopHits default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopHits("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getTopSearches method.
  //
  // get getTopSearches with minimal parameters
  void snippetForGetTopSearches() {
    // >SEPARATOR getTopSearches default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getTopSearches("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the getUsersCount method.
  //
  // get getUsersCount with minimal parameters
  void snippetForGetUsersCount() {
    // >SEPARATOR getUsersCount default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.getUsersCount("index");
    // >LOG
    // SEPARATOR<
  }

  // Snippet for the setClientApiKey method.
  //
  // switch API key
  void snippetForSetClientApiKey() {
    // >SEPARATOR setClientApiKey default
    // Initialize the client
    AnalyticsClient client = new AnalyticsClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY", "ALGOLIA_APPLICATION_REGION");

    // Call the API
    client.setClientApiKey("updated-api-key");
    // >LOG
    // SEPARATOR<
  }
}

File: snippets/javascript/src/analytics.ts
/* eslint no-console: ["error", { allow: ["log"] }] */

import { analyticsClient } from '@algolia/client-analytics';

// Snippet for the customDelete method.
//
// allow del method for a custom path with minimal parameters
export async function snippetForcustomDelete(): Promise<void> {
  // >SEPARATOR customDelete
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customDelete({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customGet method.
//
// allow get method for a custom path with minimal parameters
export async function snippetForcustomGet(): Promise<void> {
  // >SEPARATOR customGet
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customGet({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customPost method.
//
// allow post method for a custom path with minimal parameters
export async function snippetForcustomPost(): Promise<void> {
  // >SEPARATOR customPost
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customPost({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customPut method.
//
// allow put method for a custom path with minimal parameters
export async function snippetForcustomPut(): Promise<void> {
  // >SEPARATOR customPut
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customPut({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getAverageClickPosition method.
//
// get getAverageClickPosition with minimal parameters
export async function snippetForgetAverageClickPosition(): Promise<void> {
  // >SEPARATOR getAverageClickPosition
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getAverageClickPosition({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getClickPositions method.
//
// get getClickPositions with minimal parameters
export async function snippetForgetClickPositions(): Promise<void> {
  // >SEPARATOR getClickPositions
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getClickPositions({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getClickThroughRate method.
//
// get getClickThroughRate with minimal parameters
export async function snippetForgetClickThroughRate(): Promise<void> {
  // >SEPARATOR getClickThroughRate
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getClickThroughRate({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getConversationRate method.
//
// get getConversationRate with minimal parameters
export async function snippetForgetConversationRate(): Promise<void> {
  // >SEPARATOR getConversationRate
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getConversationRate({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getNoClickRate method.
//
// get getNoClickRate with minimal parameters
export async function snippetForgetNoClickRate(): Promise<void> {
  // >SEPARATOR getNoClickRate
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getNoClickRate({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getNoResultsRate method.
//
// get getNoResultsRate with minimal parameters
export async function snippetForgetNoResultsRate(): Promise<void> {
  // >SEPARATOR getNoResultsRate
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getNoResultsRate({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getSearchesCount method.
//
// get getSearchesCount with minimal parameters
export async function snippetForgetSearchesCount(): Promise<void> {
  // >SEPARATOR getSearchesCount
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getSearchesCount({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getSearchesNoClicks method.
//
// get getSearchesNoClicks with minimal parameters
export async function snippetForgetSearchesNoClicks(): Promise<void> {
  // >SEPARATOR getSearchesNoClicks
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getSearchesNoClicks({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getSearchesNoResults method.
//
// get getSearchesNoResults with minimal parameters
export async function snippetForgetSearchesNoResults(): Promise<void> {
  // >SEPARATOR getSearchesNoResults
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getSearchesNoResults({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getStatus method.
//
// get getStatus with minimal parameters
export async function snippetForgetStatus(): Promise<void> {
  // >SEPARATOR getStatus
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getStatus({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopCountries method.
//
// get getTopCountries with minimal parameters
export async function snippetForgetTopCountries(): Promise<void> {
  // >SEPARATOR getTopCountries
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopCountries({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopFilterAttributes method.
//
// get getTopFilterAttributes with minimal parameters
export async function snippetForgetTopFilterAttributes(): Promise<void> {
  // >SEPARATOR getTopFilterAttributes
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopFilterAttributes({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopFilterForAttribute method.
//
// get getTopFilterForAttribute with minimal parameters
export async function snippetForgetTopFilterForAttribute(): Promise<void> {
  // >SEPARATOR getTopFilterForAttribute
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopFilterForAttribute({
    attribute: 'myAttribute',
    index: 'index',
  });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopFiltersNoResults method.
//
// get getTopFiltersNoResults with minimal parameters
export async function snippetForgetTopFiltersNoResults(): Promise<void> {
  // >SEPARATOR getTopFiltersNoResults
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopFiltersNoResults({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopHits method.
//
// get getTopHits with minimal parameters
export async function snippetForgetTopHits(): Promise<void> {
  // >SEPARATOR getTopHits
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopHits({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getTopSearches method.
//
// get getTopSearches with minimal parameters
export async function snippetForgetTopSearches(): Promise<void> {
  // >SEPARATOR getTopSearches
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getTopSearches({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getUsersCount method.
//
// get getUsersCount with minimal parameters
export async function snippetForgetUsersCount(): Promise<void> {
  // >SEPARATOR getUsersCount
  // Initialize the client
  const client = analyticsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getUsersCount({ index: 'index' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}
